ReactAndroid/src/main/java/com/facebook/react/views/view/ReactDrawableHelper.java

```diff
@@ -41,48 +42,77 @@ public static Drawable createDrawableFromJSDescription(
         throw new JSApplicationIllegalArgumentException(
-            "Attribute " + attr + " couldn't be resolved into a drawable");
+                "Attribute " + attr + " couldn't be resolved into a drawable");
       }
+      Drawable drawable = getDefaultThemeDrawable(context);
+      return setRadius(drawableDescriptionDict, drawable);
     } else if ("RippleAndroid".equals(type)) {
-      if (Build.VERSION.SDK_INT < Build.VERSION_CODES.LOLLIPOP) {
-        throw new JSApplicationIllegalArgumentException(
-            "Ripple drawable is not available on " + "android API <21");
-      }
-      int color;
-      if (drawableDescriptionDict.hasKey(ViewProps.COLOR)
-          && !drawableDescriptionDict.isNull(ViewProps.COLOR)) {
-        color = drawableDescriptionDict.getInt(ViewProps.COLOR);
-      } else {
-        if (context
-            .getTheme()
-            .resolveAttribute(android.R.attr.colorControlHighlight, sResolveOutValue, true)) {
-          color = context.getResources().getColor(sResolveOutValue.resourceId);
-        } else {
-          throw new JSApplicationIllegalArgumentException(
-              "Attribute colorControlHighlight " + "couldn't be resolved into a drawable");
-        }
-      }
-      Drawable mask = null;
-      if (!drawableDescriptionDict.hasKey("borderless")
-          || drawableDescriptionDict.isNull("borderless")
-          || !drawableDescriptionDict.getBoolean("borderless")) {
-        mask = new ColorDrawable(Color.WHITE);
-      }
-      ColorStateList colorStateList =
-          new ColorStateList(new int[][] {new int[] {}}, new int[] {color});
-      return new RippleDrawable(colorStateList, null, mask);
+      return getRippleDrawable(context, drawableDescriptionDict);
     } else {
       throw new JSApplicationIllegalArgumentException("Invalid type for android drawable: " + type);
     }
   }
+
+  private static Drawable getDefaultThemeDrawable(Context context) {
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+      return context
+              .getResources()
+              .getDrawable(sResolveOutValue.resourceId, context.getTheme());
+    } else {
+      return context.getResources().getDrawable(sResolveOutValue.resourceId);
+    }
+  }
+
+  private static RippleDrawable getRippleDrawable(Context context, ReadableMap drawableDescriptionDict) {
+    if (Build.VERSION.SDK_INT < Build.VERSION_CODES.LOLLIPOP) {
+      throw new JSApplicationIllegalArgumentException(
+              "Ripple drawable is not available on " + "android API <21");
+    }
+    int color = getColor(context, drawableDescriptionDict);
+    Drawable mask = getMask(drawableDescriptionDict);
+    ColorStateList colorStateList =
+        new ColorStateList(new int[][] {new int[] {}}, new int[] {color});
+
+    RippleDrawable rd = new RippleDrawable(colorStateList, null, mask);
+    setRadius(drawableDescriptionDict, rd);
```

Comment:
This method not only retrieve the RippleDrawable but also set the radius? 
can you extract the call to setRadius out of this method the same way you did it in getDefaultThemeDrawable?

Reply:
ah, yes, that's cleaner. Done.